File: scripts/ssh-check.js
const { Client } = require('ssh2');

const conn = new Client();
const config = {
  host: '101.34.91.205',
  port: 22,
  username: 'ubuntu',
  password: 'Star145051',
  readyTimeout: 20000,
};

function exec(cmd) {
  return new Promise((resolve, reject) => {
    conn.exec(cmd, (err, stream) => {
      if (err) return reject(err);
      let stdout = '';
      let stderr = '';
      stream.on('close', (code) => {
        resolve({ code, stdout, stderr });
      }).on('data', (data) => {
        stdout += data.toString();
      }).stderr.on('data', (data) => {
        stderr += data.toString();
      });
    });
  });
}

conn.on('ready', async () => {
  console.log('=== SSH 连接成功 ===');
  
  try {
    let r;
    
    console.log('\n--- 系统信息 ---');
    r = await exec('uname -a && echo "---" && cat /etc/os-release | head -5 && echo "---" && whoami && echo "---" && df -h /');
    console.log(r.stdout);
    
    console.log('\n--- Node.js 版本 ---');
    r = await exec('node -v 2>&1 || echo "Node not installed"');
    console.log(r.stdout.trim());
    
    console.log('\n--- npm 版本 ---');
    r = await exec('npm -v 2>&1 || echo "npm not installed"');
    console.log(r.stdout.trim());
    
    console.log('\n--- 已开放端口 ---');
    r = await exec('ss -tlnp 2>/dev/null || netstat -tlnp 2>/dev/null || echo "no net tools"');
    console.log(r.stdout.trim());
    
    console.log('\n=== 检查完成 ===');
  } catch (e) {
    console.error('执行错误:', e.message);
  }
  
  conn.end();
});

conn.on('error', (err) => {
  console.error('SSH 连接失败:', err.message);
  process.exit(1);
});

conn.connect(config);

File: deploy/ssh_check.py
import paramiko
import sys

host = '101.34.91.205'
port = 22
username = 'ubuntu'
password = 'Star145051'

def ssh_exec(ssh, cmd):
    stdin, stdout, stderr = ssh.exec_command(cmd, timeout=30)
    out = stdout.read().decode('utf-8', errors='replace')
    err = stderr.read().decode('utf-8', errors='replace')
    return out, err

try:
    ssh = paramiko.SSHClient()
    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
    print(f'正在连接 {host}...')
    ssh.connect(host, port, username, password, timeout=20, banner_timeout=20)
    print('连接成功！')
    
    print('\n=== 系统信息 ===')
    out, _ = ssh_exec(ssh, 'uname -a')
    print(out.strip())
    
    print('\n=== 操作系统 ===')
    out, _ = ssh_exec(ssh, 'cat /etc/os-release | head -5')
    print(out.strip())
    
    print('\n=== 当前用户 ===')
    out, _ = ssh_exec(ssh, 'whoami && id')
    print(out.strip())
    
    print('\n=== Node.js ===')
    out, err = ssh_exec(ssh, 'node -v 2>&1')
    print(out.strip() or '未安装')
    
    print('\n=== npm ===')
    out, err = ssh_exec(ssh, 'npm -v 2>&1')
    print(out.strip() or '未安装')
    
    print('\n=== 已监听端口 ===')
    out, _ = ssh_exec(ssh, 'ss -tlnp 2>/dev/null || netstat -tlnp 2>/dev/null')
    print(out.strip())
    
    print('\n=== 磁盘空间 ===')
    out, _ = ssh_exec(ssh, 'df -h /')
    print(out.strip())
    
    print('\n=== 内存 ===')
    out, _ = ssh_exec(ssh, 'free -h')
    print(out.strip())
    
    ssh.close()
    print('\n检查完成！')
    
except Exception as e:
    print(f'错误: {e}')
    sys.exit(1)
